Components/Graph/Graph/Jit.hpp

```diff
@@ -0,0 +1,791 @@
+    template <class Trets, class Targs>
+    void apply(Trets ret0V, Targs arg0V);
+
+    /// @brief Apply graph
+    /// 3 inputs, 1 output
+    /// @tparam Trets
+    /// @tparam Targs
+    /// @param ret0V
+    /// @param arg0V
+    /// @param arg1V
+    /// @param arg2V
+    template <class Trets, class Targs>
+    void apply(Trets ret0V, Targs arg0V, Targs arg1V, Targs arg2V);
+
+    /// @brief Apply graph
+    /// 3 inputs, 3 outputs
+    /// @tparam Trets
+    /// @tparam Targs
+    /// @param ret0V
+    /// @param ret1V
+    /// @param ret2V
+    /// @param arg0V
+    /// @param arg1V
+    /// @param arg2V
+    template <class Trets, class Targs>
+    void apply(Trets ret0V, Trets ret1V, Trets ret2V, Targs arg0V, Targs arg1V, Targs arg2V);
+
+    /// @brief Apply graph
+    /// 3 inputs, 3+3 outputs
+    /// Pure hydro models returning also physical space velocity
+    /// @tparam TyOnerets
+    /// @tparam TyTworets
+    /// @tparam Targs
+    /// @param retTyOne0V
+    /// @param retTyOne1V
+    /// @param retTyOne2V
+    /// @param arg0V
+    /// @param arg1V
+    /// @param arg2V
+    template <class TyOnerets, class TyTworets, class Targs>
+    void apply(
+        TyOnerets retTyOne0V, TyOnerets retTyOne1V, TyOnerets retTyOne2V,
+        TyTworets retTyTwo0V, TyTworets retTyTwo1V, TyTworets retTyTwo2V,
+        Targs arg0V, Targs arg1V, Targs arg2V);
+
+    /// @brief Apply graph
+    /// 5 inputs, 5+6 outputs
+    /// Pure magneto-hydro models returning also physical space velocity
+    /// @tparam TyOnerets
+    /// @tparam TyTworets
+    /// @tparam Targs
+    /// @param retTyOne0V
+    /// @param retTyOne1V
+    /// @param retTyOne2V
+    /// @param retTyOne3V
+    /// @param retTyOne4V
+    /// @param retTyTwo0V
+    /// @param retTyTwo1V
+    /// @param retTyTwo2V
+    /// @param retTyTwo3V
+    /// @param retTyTwo4V
+    /// @param retTyTwo5V
+    /// @param arg0V
+    /// @param arg1V
+    /// @param arg2V
+    /// @param arg3V
+    /// @param arg4V
+    template <class TyOnerets, class TyTworets, class Targs>
+    void apply(
+        TyOnerets retTyOne0V, TyOnerets retTyOne1V, TyOnerets retTyOne2V,
+        TyOnerets retTyOne3V, TyOnerets retTyOne4V,
+        TyTworets retTyTwo0V, TyTworets retTyTwo1V, TyTworets retTyTwo2V,
+        TyTworets retTyTwo3V, TyTworets retTyTwo4V, TyTworets retTyTwo5V,
+        Targs arg0V, Targs arg1V, Targs arg2V, Targs arg3V, Targs arg4V);
+};
+
+template <std::uint32_t RANK>
+void Jit<RANK>::setDialects()
+{
+    mlir::func::registerAllExtensions(_registry);
+    // Add the following to include *all* MLIR Core dialects, or selectively
+    // include what you need like above. You only need to register dialects that
+    // will be *parsed* by the tool, not the one generated
+    registerAllDialects(_registry);
+    _ctx.appendDialectRegistry(_registry);
+    // Load our Dialect in this MLIR Context.
+    _ctx.loadDialect<mlir::quiccir::QuiccirDialect>();
+
+    // #ifndef NDEBUG
```

Comment:
Enable?

Reply:
I will remove it, I haven't finished the setup.